include/cachemere/policy/eviction_segmented_lru.hpp

```diff
@@ -93,19 +93,27 @@ template<class Key, class Value> void EvictionSegmentedLRU<Key, Value>::on_cache
 
 template<class Key, class Value> void EvictionSegmentedLRU<Key, Value>::on_evict(const Key& key)
 {
-    assert((m_probation_list.empty() && !m_protected_list.empty() && m_protected_list.back().get() == key) ||
-           (!m_probation_list.empty() && m_probation_list.back().get() == key));
+    assert((m_probation_list.empty() && !m_protected_list.empty()) || !m_probation_list.empty());
```

Comment:
you can just keep the part `!m_protected_list.empty()) || !m_probation_list.empty()`